pydis_site/apps/api/viewsets/bot/offensive_message.py

```diff
@@ -0,0 +1,66 @@
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework.filters import SearchFilter
+from rest_framework.mixins import (
+    CreateModelMixin,
+    DestroyModelMixin,
+    ListModelMixin,
+    UpdateModelMixin
+)
+from rest_framework.viewsets import GenericViewSet
+
+from pydis_site.apps.api.models.bot.offensive_message import OffensiveMessage
+from pydis_site.apps.api.serializers import OffensiveMessageSerializer
+
+
+class OffensiveMessageViewSet(
+    CreateModelMixin, ListModelMixin, DestroyModelMixin, UpdateModelMixin, GenericViewSet
+):
+    """
+    View providing CRUD access to reminders.
+
+    ## Routes
+    ### GET /bot/offensive-message
+    Returns all offensive messages in the database.
+
+    #### Response format
+    >>> [
+    ...     {
+    ...         'id': '631953598091100200',
+    ...         'channel_id': '291284109232308226',
+    ...         'delete_date': '2020-01-01'
+    ...     },
+    ...     ...
+    ... ]
+
+    #### Status codes
+    - 200: returned on success
+
+    ### POST /bot/offensive-message
+    Create a new offensive message object.
+
+    #### Request body
+    >>> {
+    ...     'id': int,
+    ...     'channel_id': int,
+    ...     'delete_date': datetime.date  # ISO-8601-formatted date
+    ... }
+
+    #### Status codes
+    - 201: returned on success
+    - 400: if the body format is invalid
+
+    ### DELETE /bot/offensive-message/<id:int>
+    Delete the offensive message object with the given `id`.
+
+    #### Status codes
+    - 204: returned on success
+    - 404: if a offensive message object with the given `id` does not exist
+
+    ## Authentication
+    Requires an API token.
```

Comment:
Are you planning to use this endpoint to delete the entries after the corresponding messages have been successfully removed from the guild? I don't think I've seen it in use in the `bot` PR, but I've only taken a cursory view at that one. If we're not going to use it, I don't think we should expose a delete method.